### DISCUSSION TITLE
`cluster` should accept a callback instead of size argument

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#299] [Ideas]
This allows `cluster` to be used similarly to `group`

```ts
import * as _ from 'radashi'

const data = [1, 2, 3]

const groups = _.group(data, n => n % 2)
// => { 0: [2], 1: [1, 3] }

const clusters = _.cluster(data, n => n % 2)
// => [[1, 3], [2]]
```

- **Why not do `Object.values(_.group(…))`?**  
  Because that results in `(number[] | undefined)[]` and there's nothing we can do about that, except for a manual type assertion (which is error-prone).

Inspired by #287

**Comment by Minhir** [comment]
It totally breaks `cluster` semantics (BTW, I don't really like cluster naming. `chunk` sounds much better to me). We can introduce `groupArr` or some different name for it.

**Reply by aleclarson** [reply]
`chunk` makes sense, and Lodash uses it. 👍 

I agree we should keep `group` in the name, but my rationale for that is different than yours: to make it easily discoverable and clearly related to `group`.

I don't fully understand the idea that it breaks `cluster`'s semantics, since clustering/chunking is a form of grouping, but since we agree, there's no point in debating it. 😃
